Documentation/contributing/development/reviewers_committers/review_vendor.rst

```diff
@@ -0,0 +1,117 @@
+=======================
+
+This section describes some of the process and expectations for reviewing PRs
+on behalf of @cilium/vendor. Note that :ref:`the generic PR review
+process for Committers <review_process>` still applies, even though it is not
+specific to dependencies.
+
+Existing Dependencies
+---------------------
+
+Updates to existing dependencies most commonly occur through PRs opened by
+`Renovate <renovate_>`_, which is a 3rd party service used throughout the
+Cilium organization. Renovate continually checks repositories for out-of-date
+dependencies and opens new PRs to update any it finds.
+
+.. note::
+   Renovate is configured via a file named ``renovate.json5`` found within the
+   ``.github/`` folder of each repository it is enabled on. Cilium's Renovate
+   configuration is long and complex, having been developed to enable
+   dependency updates to Docker images and GitHub Actions. Only a few pieces
+   of the configuration are relevant to the @cilium/vendor team.
+
+.. _renovate: https://docs.renovatebot.com
+
+When reviewing PRs that update an existing dependency, members of the
+@cilium/vendor team are required to ensure that the update does not include
+any breaking changes or licensing issues. These checks are facilitated via
+GitHub Action CI workflows, which are triggered by commenting ``/test`` within
+a PR. See :ref:`CI  / GitHub Actions <ci_gha>` for more information on their
+use.
+
+New Dependencies
+----------------
+
+When a new dependency is added as part of a PR, the @cilium/vendor team will
+be assigned to ensure the new dependency meets the following criteria:
+
+1. The new dependency must add functionality that is not already provided within
+   an existing dependency, an internal package to the project, or Go's
+   standard library.
+2. The functionality provided by the new dependency must be non-trivial to
+   re-implement manually.
+3. The new dependency must be actively maintained, having new commits and/or
+   releases within the past year.
+4. The new dependency must appear to be of generally good quality.
+
+These criteria ensure the long-term success of the project by justifying the
+inclusion of the new dependency into the project's codebase.
+
+Cilium Imports
+--------------
+
+A subset of the repositories the @cilium/vendor team is responsible for import
+code from cilium/cilium as a dependency. A complication in this relationship
+is the usage of `replace directives <replace_directives_>`_ in the
+`cilium/cilium go.mod file <go_dot_mod_>`_. Replace directives are only applied
+to the go.mod file they are used in and do not carry over when imported by
+another modules. This creates the need for replace directives used in
+the cilium/cilium go.mod file to by synced with any module which imports
+cilium/cilium as a dependency.
+
+The vendor team is therefore responsible for ensuring that replace directives
+are synced when reviewing PRs which update the version of a cilium/cilium
+dependency.
```

Comment:
Maybe to clarify, this specific paragraph I highlighted - is this saying that as a vendor codeowner you need to follow up that the repository behind the `replace` directive is updated before approving a PR? Or it's specifically the line for replace that also needs to be updated? Maybe if there was an example diff that shows what changes must happen at the same time, it would serve as a good example about how a reviewer would recognize this pattern and respond with the expected review.

(Maybe also if we can detect this with scripting/tooling then it'll shift the responsibility more into CI and to the developer rather than relying on vendor review to catch... but I don't know what that would involve. Thinking out loud, you don't need to resolve that in this PR.)

Reply:
That's a good point. I wasn't sure on where the vendor team's responsibility lies in deciding to add a new `replace` directive. It seems like more of a case-by-case scenario to address a bugfix that can't/won't be upstreamed.

I think the difficult part in removing a replace directive and instead referencing the fork directly is that this would only apply to imports in the cilium codebase and not to any other dependencies that may also import the bugged dependency.

Maybe we are at a point where this isn't a problem though, I'm not sure.

Reply:
Perhaps others from @cilium/vendor could chime in about their experiences with the `replace` directive. I know it's created some pain points but I don't know well enough to suggest how we might write this section to help vendor team members to apply a consistent bar.

Reply:
> I wasn't sure on where the vendor team's responsibility lies in deciding to add a new replace directive. It seems like more of a case-by-case scenario to address a bugfix that can't/won't be upstreamed.

I don't think it should be the vendor team's responsibility to definitely decide on whether to add a `replace` directive or not. There might be legitimate cases, like Joe pointed out above:

> In practice I think it's used relatively sparsely today, mainly where we've been forced to carry our own fork of an upstream library. But then maybe we should just be moving Cilium to directly reference the fork if there's no path to upstream those changes.

In my recent experience, `replace` directives were fairly short lived when a change needed in an upstream library wasn't released yet. In other cases where there are non-trivial, non-upstreamable changes we've changed the imports to use the fork directly, e.g. as done for `cilium/dns` in https://github.com/cilium/cilium/pull/27582.

As a general guideline I'd suggest the following to be documented:

* Explicitly discourage the use of `replace` directives
* If there is no other alternative to a `replace` directive (such as pending upstream changes) try to remove it before a Cilium release. It might be good to create a release blocker issue and assign it to the developer adding the `replace` directive.
* Require an accompanying comment to the directive stating the reason for it to be added, as is e.g. the case for `sigs.k8s.io/controller-tools`: https://github.com/cilium/cilium/blob/aa6f482d96b3e4989b28b74e89fa4ff798f84624/go.mod#L285-L287

Reply:
I adjusted the wording based on the thoughts in this thread. Let me know what you think, thanks!